### PR TITLE
refactor: update jobs to run on latest os version

### PR DESCRIPTION
**Pull request by kov117**

### CODE REVIEW

**Comment by hknutsen** [comment]
@kov117 The Terraform workflow uses the [`hashicorp/setup-terraform`](https://github.com/marketplace/actions/hashicorp-setup-terraform) action to install Terraform instead of relying of the version that used to be pre-installed on GitHub runners, so we should be fine, but feel free to test before getting an approval.

**Comment by hknutsen** [comment]
Merged changes from main branch, which contains fixes for linter errors you were having in this branch 🕺

**Comment by kov117** [comment]
@equinor/ops-actions-maintainers test was succesfull, it was as @hknutsen said, everything turned out fine. So review away 👍🏼

**Review by hknutsen — APPROVED** [review]

**Pull request merged**
